### PR TITLE
Fix docs

### LINKED ISSUE #200: Review class docstrings for refactored modules
The docstring for this class is woe-fully out of date.

### PR DESCRIPTION
**Pull request by klauer**
A once-over, not comprehensive. Docs at least build locally. Docstrings are more accurate than before.

Used pydocstyle to help keep things consistent in several modules. Overall, it's a good tool but a bit painful to adhere to at times.

Closes #200

### CODE REVIEW

**Comment by klauer** [comment]
@ZLLentz this may have some conflicts with #291. We can let that go in first and I'll deal with rebasing/merging here.

File: typhos/display.py [hunk 1]
```diff
@@ -224,18 +284,20 @@ def handle_menu(checked):
...
+            Embedded
+            Detailed
+            Engineering
+            -------------
+            Refresh templates
+            -------------
+            Kind filter > Show hinted
+                          ...
+                          Show only hinted
+            Filter by name
```
Comment (hhslepicka) [inline typhos/display.py:300]: I forgot that when I added this option.
```suggestion
            Filter by name
            Hide Empty Panels
```

File: typhos/display.py [hunk 2]
```diff
@@ -277,6 +340,7 @@ def __init__(self, display_type, *, parent=None):
         self.templates = None
 
     def _clicked(self):
+        """Clicked callback - set thet template."""
```
Comment (hhslepicka) [inline typhos/display.py:343]:
```suggestion
        """Clicked callback - set the template."""
```

File: typhos/display.py [hunk 3]
```diff
@@ -388,9 +461,22 @@ def mousePressEvent(self, event):
 
 class TyphosDisplayTitle(QtWidgets.QFrame, widgets.TyphosDesignerMixin):
     """
-    Standardized Typhos Device Display title
+    Standardized Typhos Device Display title.
+
+    Parameters
+    ----------
+    title : str, optional
+        The initial title text, which may contain macros.
```
Comment (hhslepicka) [inline typhos/display.py:469]: Should we mention that it defaults to the `${name}` macro?

File: typhos/display.py [hunk 4]
```diff
@@ -1069,6 +1178,7 @@ def toggle_display(widget, force_state=None):
     ----------
     widget : QWidget
         The widget in which to look for Panels
```
Comment (hhslepicka) [inline typhos/display.py:1181]:
```suggestion
        The widget in which to look for Panels.
```

File: typhos/display.py [hunk 5]
```diff
@@ -1037,7 +1145,7 @@ def from_class(cls, klass, *, template=None, macros=None, **kwargs):
     @classmethod
     def _get_specific_screens(cls, device_cls):
         """
-        Get the list of specific screens for a given device class
+        Get the list of specific screens for a given device class.
 
         That is, screens that are not default Typhos-provided screens
```
Comment (hhslepicka) [inline typhos/display.py:1151]:
```suggestion
        That is, screens that are not default Typhos-provided screens.
```

File: typhos/display.py [hunk 6]
```diff
@@ -878,6 +982,7 @@ def add_device(self, device, macros=None):
         ----------
         device: ophyd.Device
             The device to add
```
Comment (hhslepicka) [inline typhos/display.py:985]:
```suggestion
            The device to add.
```

File: typhos/display.py [hunk 7]
```diff
@@ -990,8 +1093,10 @@ def from_device(cls, device, template=None, macros=None, **kwargs):
 
         template :str, optional
             Set the ``display_template``
```
Comment (hhslepicka) [inline typhos/display.py:1096]:
```suggestion
            Set the ``display_template``.
```

File: typhos/display.py [hunk 8]
```diff
@@ -990,8 +1093,10 @@ def from_device(cls, device, template=None, macros=None, **kwargs):
...
+
         macros: dict, optional
             Macro substitutions to be placed in template
```
Comment (hhslepicka) [inline typhos/display.py:1099]:
```suggestion
            Macro substitutions to be placed in template.
```

File: typhos/display.py [hunk 9]
```diff
@@ -990,8 +1093,10 @@ def from_device(cls, device, template=None, macros=None, **kwargs):
...
+
         **kwargs
             Passed to the class init
```
Comment (hhslepicka) [inline typhos/display.py:1102]:
```suggestion
            Passed to the class init.
```

File: typhos/display.py [hunk 10]
```diff
@@ -1006,18 +1111,21 @@ def from_device(cls, device, template=None, macros=None, **kwargs):
...
 
         Loads the signals in to the appropriate positions and sets the title to
         a cleaned version of the device name
 
         Parameters
         ----------
         klass : str or class
+
         template :str, optional
             Set the ``display_template``
```
Comment (hhslepicka) [inline typhos/display.py:1125]:
```suggestion
            Set the ``display_template``.
```

File: typhos/display.py [hunk 11]
```diff
@@ -1006,18 +1111,21 @@ def from_device(cls, device, template=None, macros=None, **kwargs):
...
+
         macros: dict, optional
             Macro substitutions to be placed in template
```
Comment (hhslepicka) [inline typhos/display.py:1128]:
```suggestion
            Macro substitutions to be placed in template.
```

File: typhos/display.py [hunk 12]
```diff
@@ -1006,18 +1111,21 @@ def from_device(cls, device, template=None, macros=None, **kwargs):
...
+
         **kwargs
             Extra arguments are used at device instantiation
```
Comment (hhslepicka) [inline typhos/display.py:1131]:
```suggestion
            Extra arguments are used at device instantiation.
```

File: typhos/panel.py [hunk 13]
```diff
@@ -99,7 +144,14 @@ def __init__(self, signals=None):
...
     @property
     def signals(self):
-        """Return all instantiated signals, omitting components."""
+        """
+        Get all instantiated signals, omitting components.
+
+        Returns
+        -------
+        signals : dict
+            With the form: ``{signal_name: signal}``
```
Comment (hhslepicka) [inline typhos/panel.py:153]:
```suggestion
            With the form: ``{signal_name: signal}``.
```

File: typhos/panel.py [hunk 14]
```diff
@@ -108,28 +160,40 @@ def signals(self):
...
     @property
     def visible_signals(self):
-        """Return all visible signals, omitting components."""
+        """
+        Get all signals visible according to filters, omitting components.
+
+        Returns
+        -------
+        signals : dict
+            With the form: ``{signal_name: signal}``
```
Comment (hhslepicka) [inline typhos/panel.py:169]:
```suggestion
            With the form: ``{signal_name: signal}``.
```

File: typhos/panel.py [hunk 15]
```diff
@@ -180,18 +244,24 @@ def _create_row_label(self, attr, dotted_name, tooltip):
...
 
+        If widget information for the given signal is available in the global
+        cache, the widgets will be created immediately.  Otherwise, a row will
+        be reserved and widgets created upon signal connection and background
+        description callback.
+
         Parameters
         ----------
         signal : EpicsSignal, EpicsSignalRO
             Signal to create a widget
```
Comment (hhslepicka) [inline typhos/panel.py:260]:
```suggestion
            Signal to create a widget.
```

File: typhos/panel.py [hunk 16]
```diff
@@ -291,10 +380,12 @@ def _update_row(self, row, widgets, **kwargs):
         ----------
         row : int
             The row number
```
Comment (hhslepicka) [inline typhos/panel.py:382]:
```suggestion
            The row number.
```

File: typhos/panel.py [hunk 17]
```diff
@@ -392,12 +521,13 @@ def _set_visible(self, signal_name, visible):
...
+        Filter signals based on the given kinds.
 
         Parameters
         ----------
         kinds : list of :class:`ophyd.Kind`
-            If given
+            List of kinds to show.
+
         name_filter : str, optional
             Additionally filter signals by name
```
Comment (hhslepicka) [inline typhos/panel.py:532]:
```suggestion
            Additionally filter signals by name.
```

File: typhos/panel.py [hunk 18]
```diff
@@ -449,10 +578,13 @@ def _maybe_add_signal(self, device, attr, dotted_name, component):
...
             The device owner
+
         attr : str
             The signal's attribute name
+
         dotted_name : str
             The signal's dotted name
+
         component : ophyd.Component
             The component class used to generate the instance
```
Comment (hhslepicka) [inline typhos/panel.py:589]:
```suggestion
            The device owner.

        attr : str
            The signal's attribute name.

        dotted_name : str
            The signal's dotted name.

        component : ophyd.Component
            The component class used to generate the instance.
```

**Pull request merged**